### PR TITLE
Use the new Dart 2.3.0 list literal syntax

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

**Comment by nex3** [comment]
@munificent I owe you a drink for this feature!

source

File: lib/src/ast/css/media_query.dart [hunk 1]
```diff
@@ -54,8 +54,8 @@ class CssMediaQuery {
     var theirType = other.type?.toLowerCase();
 
     if (ourType == null && theirType == null) {
-      return MediaQuerySuccessfulMergeResult._(CssMediaQuery.condition(
-          this.features.toList()..addAll(other.features)));
+      return MediaQuerySuccessfulMergeResult._(
+          CssMediaQuery.condition([...this.features, ...other.features]));
```
Comment (jelbourn) [inline lib/src/ast/css/media_query.dart]: MRW seeing JavaScript syntax array literals with destructing in Dart

its-a-unix-system

**Comment by nex3** [comment]
@jelbourn PTAL; I found a bunch more places that could be updated

File: lib/src/ast/sass/argument_declaration.dart [hunk 2]
```diff
@@ -100,9 +100,8 @@ class ArgumentDeclaration implements SassNode {
...
 
-  String toString() {
-    var components = List.of(arguments.map((arg) => arg.toString()));
-    if (restArgument != null) components.add('$restArgument...');
-    return components.join(', ');
-  }
+  String toString() => [
+        for (var arg in arguments) arg.toString(),
+        if (restArgument != null) '$restArgument...'
+      ].join(', ');
```
Comment (munificent) [inline lib/src/ast/sass/argument_declaration.dart]: image

**Comment by leafpetersen** [comment]
This is awesome.

File: lib/src/ast/css/media_query.dart (hunk 1 above)
Comment (mraleph) [reply lib/src/ast/css/media_query.dart]: fwiw this is not destructuring though, this is spread. Dart does not have destructuring.

**Pull request merged**